### DISCUSSION TITLE
Best way to debug 'unable to connect to the router' when clicking the ping button

### DISCUSSION THREAD

**Discussion by amulheirn** [peering-manager/peering-manager#727] [Q&A]
Sorry if this isnt the right place to ask for help, but I thought this would be worth a try:

I have two routers configured in Peering Manager - one is Junos, the other is Nokia SROS.   I can successfully connect to the Junos device when I click ping in the devices screen, but cannot connect to the SROS device.

I have followed the instructions [here](https://github.com/napalm-automation-community/napalm-sros) for installing the Napalm driver for SROS, and I can successfully connect to the Nokia device from Python.

I have Peering Manager installed, checked out to release v1.7.4.  

I have added the routers under Devices | Routers - one has platform = junos, the other has platform = sros.

In configuration.py I have a Napalm user, password and the following args set:  `NAPALM_ARGS = {'port':22}`.  This is necessary because our management firewall only permits port 22.

I have created an 'sros' platform in the GUI, but the password algorithm field only presents Juniper or Cisco options.  So in the device, I have unchecked the 'encrypt passwords' field.

From the underlying linux host where Peering Manager is installed, I can ssh to both devices ok.  I can also ssh with '-s netconf' and I get a netconf response from both devices.  So I am confident there is no firewall issue between the Peering Manager host and the device.

However, when I click the ping button on the SROS device in Peering Manager, I get 'unable to connect to the router'.    

If I run tcpdump -i eth0 host <host ip> I see no packets going out to the SROS device at all.   If I do the same for the Junos device, tcpdump captures packets destined to port 22.   So it seems as though Napalm can connect in native python, but Peering Manager cannot - it doesn't even send any packets out.

What is the best way to debug this?  The logs directory does not seem to show any information other than a PeeringDB log file.

Any help would be much appreciated!

Andy

**Comment by amulheirn** [comment]
A small update and progress on this.

I did some manual reading and enabled Napalm logs.   The result is this:

```@peering-manager:/opt/peering-manager/logs]$ more napalm.log 
2023-05-25 20:52:21 | DEBUG | looking for napalm driver 'sros'
2023-05-25 20:52:21 | ERROR | no napalm driver: 'sros' for platform: 'sros' found (not installed or does not exist)
2023-05-25 20:52:21 | DEBUG | testing connection with 10.6.1.3
2023-05-25 20:52:21 | ERROR | cannot connect to 10.6.1.3, napalm functions won't work
```

So it seems that the napalm driver name I have set in Peering Manager is not right.   But the driver name I use in Python is 'sros' so I need to work out why this is not working.

**Comment by amulheirn** [comment]
What is odd is that in the napalm-sros install README, it says the driver is called 'sros' - and this driver name works fine in Python:
```
from napalm import get_network_driver
driver = get_network_driver("sros")
```

**Comment by gmazoyer** [comment]
Are you sure that the `sros` driver for NAPALM is installed in the Peering Manager (virtual) environment? The error you are facing usually comes from that.

**Reply by netravnen** [reply]
Maybe add a section to the docs concerning this bullet [point](https://peering-manager.readthedocs.io/en/latest/setup/3-peering-manager/)?

```
(venv) # echo 'gunicorn' >> local_requirements.txt
(venv) # pip3 install -r local_requirements.txt

```
Somewhere else add this mention

```
(venv) # echo 'napalm-sros' >> local_requirements.txt
(venv) # pip3 install -r local_requirements.txt
```

@amulheirn If you add `napalm-sros` to `local_requirements.txt`. The PyPi package will be automatically installed (and upgraded) when installing (and upgrading) peering-manager. ;-) -- Remember: The Python Virtual Environment is _always_ wiped/erased when using the standard peering-mananger upgrade script. Non-standard (not in the usual requirements file) needs to be explicitly mentioned in the local_requirements.txt file instead to be installed.
